core/google/cloud/iam.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Non-API-specific IAM policy definitions
+
+For allowed roles / permissions, see:
+https://cloud.google.com/iam/docs/understanding-roles
+"""
+
+# Generic IAM roles
+
+OWNER_ROLE = 'roles/owner'
+"""Generic role implying all rights to an object."""
+
+EDITOR_ROLE = 'roles/editor'
+"""Generic role implying rights to modify an object."""
+
+VIEWER_ROLE = 'roles/viewer'
+"""Generic role implying rights to access an object."""
+
+
+class Policy(object):
+    """IAM Policy
+
+    See:
+    https://cloud.google.com/iam/reference/rest/v1/Policy
+
+    :type etag: str
+    :param etag: ETag used to identify a unique of the policy
+
+    :type version: int
+    :param version: unique version of the policy
+    """
+    _OWNER_ROLES = (OWNER_ROLE,)
+    """Roles mapped onto our ``owners`` attribute."""
+
+    _EDITOR_ROLES = (EDITOR_ROLE,)
+    """Roles mapped onto our ``editors`` attribute."""
+
+    _VIEWER_ROLES = (VIEWER_ROLE,)
+    """Roles mapped onto our ``viewers`` attribute."""
+
+    def __init__(self, etag=None, version=None):
+        self.etag = etag
+        self.version = version
+        self.bindings = {}
+
+    @property
+    def owners(self):
+        """Legacy access to owner role."""
+        result = set()
+        for role in self._OWNER_ROLES:
+            for member in self.bindings.get(role, ()):
+                result.add(member)
+        return result
```

Comment:
Ahhh yuk. Ok. How can the `owners` property support set modification syntax like it did previously? In fact, how did the previous version support set modification? How did it know which role to attribute an addition to 'owners' to? This seems really bad as it makes policies opaque to client library users.

As an aside. Set modification is much more idiomatic than reassignment since this API follows a 
read-modify-write pattern with the `etag`.